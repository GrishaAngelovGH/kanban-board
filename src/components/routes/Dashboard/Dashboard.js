import Page from "components/routes/Page"

import TasksPerMemberCard from "./Cards/TasksPerMemberCard"
import TasksPerColumnCard from "./Cards/TasksPerColumnCard"
import TasksPerPriorityCard from "./Cards/TasksPerPriorityCard"

const Dashboard = () => {
  const description = (
    <div className="col-10">
      <h3 className="m-0 text-center">Dashboard</h3>
    </div>
  )

  return (
    <Page description={description}>
      <div className="row g-0 mt-3 justify-content-center overflow-auto tasks">
        <div className="col-10">
          <div className="row mb-5 gap-5 justify-content-between">
            <TasksPerMemberCard />
            <TasksPerColumnCard />
          </div>
          <div className="row mb-5 gap-5 justify-content-between">
            <TasksPerPriorityCard />
          </div>
        </div>
      </div>
    </Page>
  )
}

export default Dashboard