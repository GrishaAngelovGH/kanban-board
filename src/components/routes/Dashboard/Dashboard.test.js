import { render } from "@testing-library/react"

import {
  BrowserRouter as Router,
  Routes,
  Route
} from "react-router-dom"

import Dashboard from "./Dashboard"

jest.mock('react-chartjs-2', () => ({ Bar: () => <div>Bar Chart</div> }))

test("should render Dashboard route component", () => {
  const view = render(
    <Router>
      <Routes>
        <Route path="/" element={<Dashboard />} />
      </Routes>
    </Router>
  )

  expect(view).toMatchSnapshot()
})