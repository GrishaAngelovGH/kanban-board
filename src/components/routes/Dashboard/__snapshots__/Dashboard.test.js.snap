// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dashboard route component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row g-0 min-vh-100"
        style="background-image: url(geometric-triangle-shapes-background.jpg); background-size: cover;"
      >
        <div
          class="col-12"
        >
          <div
            class="row g-0 bg-light p-2 shadow align-items-center"
          >
            <div
              class="col-2 col-md-1"
            >
              <a
                class="btn btn-light bi bi-arrow-left border"
                href="/"
              />
            </div>
            <div
              class="col-10"
            >
              <h3
                class="m-0 text-center"
              >
                Dashboard
              </h3>
            </div>
          </div>
          <div
            class="row g-0 mt-3 justify-content-center overflow-auto tasks"
          >
            <div
              class="col-10"
            >
              <div
                class="row mb-5 gap-5 justify-content-between"
              >
                <div
                  class="col col-md-5 bg-white rounded shadow"
                  style="height: 350px;"
                >
                  <div
                    class="d-flex justify-content-between align-items-center"
                  >
                    <p
                      class="m-0 ms-2"
                    >
                      Tasks Per Member Card
                    </p>
                    <div
                      class="d-flex align-items-center"
                    >
                      <i
                        class="bi bi-star fs-4 text-warning"
                      />
                      <i
                        class="bi bi-x fs-2"
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex justify-content-center"
                  >
                    <div>
                      Bar Chart
                    </div>
                  </div>
                </div>
                <div
                  class="col col-md-5 bg-white rounded shadow"
                  style="height: 350px;"
                >
                  <div
                    class="d-flex justify-content-between align-items-center"
                  >
                    <p
                      class="m-0 ms-2"
                    >
                      Tasks Per Column Card
                    </p>
                    <div
                      class="d-flex align-items-center"
                    >
                      <i
                        class="bi bi-star fs-4 text-warning"
                      />
                      <i
                        class="bi bi-x fs-2"
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex justify-content-center"
                  >
                    <div>
                      Bar Chart
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="row mb-5 gap-5 justify-content-between"
              >
                <div
                  class="col col-md-5 bg-white rounded shadow"
                  style="height: 350px;"
                >
                  <div
                    class="d-flex justify-content-between align-items-center"
                  >
                    <p
                      class="m-0 ms-2"
                    >
                      Tasks Per Priority Card
                    </p>
                    <div
                      class="d-flex align-items-center"
                    >
                      <i
                        class="bi bi-star fs-4 text-warning"
                      />
                      <i
                        class="bi bi-x fs-2"
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex justify-content-center"
                  >
                    <div>
                      Bar Chart
                    </div>
                  </div>
                </div>
                <div
                  class="col col-md-5 bg-white rounded shadow"
                  style="height: 350px;"
                >
                  <div
                    class="d-flex justify-content-between align-items-center"
                  >
                    <p
                      class="m-0 ms-2"
                    >
                      Tasks Activity
                    </p>
                    <div
                      class="d-flex align-items-center"
                    >
                      <i
                        class="bi bi-star fs-4 text-warning"
                      />
                      <i
                        class="bi bi-x fs-2"
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex justify-content-center"
                  >
                    <div>
                      Bar Chart
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row g-0 min-vh-100"
      style="background-image: url(geometric-triangle-shapes-background.jpg); background-size: cover;"
    >
      <div
        class="col-12"
      >
        <div
          class="row g-0 bg-light p-2 shadow align-items-center"
        >
          <div
            class="col-2 col-md-1"
          >
            <a
              class="btn btn-light bi bi-arrow-left border"
              href="/"
            />
          </div>
          <div
            class="col-10"
          >
            <h3
              class="m-0 text-center"
            >
              Dashboard
            </h3>
          </div>
        </div>
        <div
          class="row g-0 mt-3 justify-content-center overflow-auto tasks"
        >
          <div
            class="col-10"
          >
            <div
              class="row mb-5 gap-5 justify-content-between"
            >
              <div
                class="col col-md-5 bg-white rounded shadow"
                style="height: 350px;"
              >
                <div
                  class="d-flex justify-content-between align-items-center"
                >
                  <p
                    class="m-0 ms-2"
                  >
                    Tasks Per Member Card
                  </p>
                  <div
                    class="d-flex align-items-center"
                  >
                    <i
                      class="bi bi-star fs-4 text-warning"
                    />
                    <i
                      class="bi bi-x fs-2"
                    />
                  </div>
                </div>
                <div
                  class="d-flex justify-content-center"
                >
                  <div>
                    Bar Chart
                  </div>
                </div>
              </div>
              <div
                class="col col-md-5 bg-white rounded shadow"
                style="height: 350px;"
              >
                <div
                  class="d-flex justify-content-between align-items-center"
                >
                  <p
                    class="m-0 ms-2"
                  >
                    Tasks Per Column Card
                  </p>
                  <div
                    class="d-flex align-items-center"
                  >
                    <i
                      class="bi bi-star fs-4 text-warning"
                    />
                    <i
                      class="bi bi-x fs-2"
                    />
                  </div>
                </div>
                <div
                  class="d-flex justify-content-center"
                >
                  <div>
                    Bar Chart
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row mb-5 gap-5 justify-content-between"
            >
              <div
                class="col col-md-5 bg-white rounded shadow"
                style="height: 350px;"
              >
                <div
                  class="d-flex justify-content-between align-items-center"
                >
                  <p
                    class="m-0 ms-2"
                  >
                    Tasks Per Priority Card
                  </p>
                  <div
                    class="d-flex align-items-center"
                  >
                    <i
                      class="bi bi-star fs-4 text-warning"
                    />
                    <i
                      class="bi bi-x fs-2"
                    />
                  </div>
                </div>
                <div
                  class="d-flex justify-content-center"
                >
                  <div>
                    Bar Chart
                  </div>
                </div>
              </div>
              <div
                class="col col-md-5 bg-white rounded shadow"
                style="height: 350px;"
              >
                <div
                  class="d-flex justify-content-between align-items-center"
                >
                  <p
                    class="m-0 ms-2"
                  >
                    Tasks Activity
                  </p>
                  <div
                    class="d-flex align-items-center"
                  >
                    <i
                      class="bi bi-star fs-4 text-warning"
                    />
                    <i
                      class="bi bi-x fs-2"
                    />
                  </div>
                </div>
                <div
                  class="d-flex justify-content-center"
                >
                  <div>
                    Bar Chart
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
